src/main/java/com/sparta/outsourcing/domain/order/repository/OrderRepository.java

```diff
@@ -3,12 +3,19 @@
 import com.sparta.outsourcing.domain.order.entity.Order;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.domain.Slice;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 
+import java.util.Optional;
+
 public interface OrderRepository extends JpaRepository<Order, Long> {
     @Query(value = "SELECT * FROM orders " +
             "WHERE is_deleted = false AND user_id = :loginUserId", nativeQuery = true)
     Slice<Order> findUserOrderHistory(@Param("loginUserId") Long loginUserId, Pageable pageable);
+
+    @Query(value = "SELECT o FROM Order AS o WHERE o.orderId = :id")
+    @EntityGraph(attributePaths = {"user", "menu"})
+    Optional<Order> findByIdEntityGraph(@Param("id") Long id);
```

Comment:
```suggestion
    @Query(value = "SELECT o FROM Order AS o WHERE o.orderId = :id")
    @EntityGraph(attributePaths = {"owner", "store", "menu"})
    Optional<Order> findByIdEntityGraph(@Param("id") Long id);
```
찾아보니 외래 키 컬럼 이름으로 해야 하는 것 같습니다.
menu에는 user가 없어서 store도 거쳐서 찾아야 하고요.

Reply:
그리고 여러 개 하면 어떻게 동작하는지 궁금해서 그런데, 쿼리를 어떻게 날리는 지 꼭 보고 싶습니다.